timelock-agent/src/main/java/com/palantir/timelock/paxos/PaxosAgent.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.paxos;
+
+import java.util.function.Consumer;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Preconditions;
+import com.palantir.atlasdb.http.BlockingTimeoutExceptionMapper;
+import com.palantir.atlasdb.http.NotCurrentLeaderExceptionMapper;
+import com.palantir.atlasdb.timelock.TimeLockServices;
+import com.palantir.atlasdb.timelock.TooManyRequestsExceptionMapper;
+import com.palantir.atlasdb.timelock.paxos.ManagedTimestampService;
+import com.palantir.atlasdb.timelock.paxos.PaxosResource;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+import com.palantir.lock.RemoteLockService;
+import com.palantir.remoting2.config.ssl.SslSocketFactories;
+import com.palantir.timelock.TimeLockAgent;
+import com.palantir.timelock.config.ImmutablePaxosRuntimeConfiguration;
+import com.palantir.timelock.config.ImmutableTimeLockDeprecatedConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration;
+import com.palantir.timelock.config.PaxosRuntimeConfiguration;
+import com.palantir.timelock.config.TimeLockDeprecatedConfiguration;
+import com.palantir.timelock.config.TimeLockInstallConfiguration;
+import com.palantir.timelock.config.TimeLockRuntimeConfiguration;
+
+import io.reactivex.Observable;
+
+public class PaxosAgent extends TimeLockAgent {
+    private final PaxosInstallConfiguration paxosInstall;
+
+    private final PaxosResource paxosResource;
+    private final PaxosLeadershipCreator leadershipAgent;
+    private final LockCreator lockCreator;
+    private final PaxosTimestampCreator timestampCreator;
+
+    public PaxosAgent(TimeLockInstallConfiguration install,
+            Observable<TimeLockRuntimeConfiguration> runtime,
+            Consumer<Object> registrar) {
+        this(install, runtime, ImmutableTimeLockDeprecatedConfiguration.builder().build(), registrar);
+    }
+
+    public PaxosAgent(TimeLockInstallConfiguration install,
+            Observable<TimeLockRuntimeConfiguration> runtime,
+            TimeLockDeprecatedConfiguration deprecated,
+            Consumer<Object> registrar) {
+        super(install, runtime, deprecated, registrar);
+        Preconditions.checkState(install.algorithm() instanceof PaxosInstallConfiguration,
```

Comment:
We can, though `PaxosInstallConfiguration` is only an algorithm configuration and we need details (e.g. cluster, security). We could take _both_ and use the `paxosInstall` directly, though that would mean that people can pass two pieces of different configs around.